packaging/irods_rest.py

```diff
@@ -1,67 +1,108 @@
 import os, pycurl, getopt, sys, urllib
 from functools import partial
-from StringIO import StringIO
+from io import StringIO ## for Python 3
 import base64
 import tempfile
 
-class old_pycurl_mock_StringIO(file):
-    def __init__(self,*a,**kw):
-        if not a:
-            self.__name = tempfile.mktemp()
-            a = (self.__name,'w+b')
-        super(old_pycurl_mock_StringIO,self).__init__(*a,**kw)
-    def __del__(self):
-        self.close()
-        os.unlink(self.__name)
-    def getvalue(self):
-        self.seek(0)
-        return self.read()
-
-class old_pycurl_mock_BytesIO(old_pycurl_mock_StringIO):
-    def __init__(self,bytestr=''):
-        super(old_pycurl_mock_BytesIO,self).__init__()
-        if bytestr:
-            self.write(bytestr)
-            self.flush()
-            self.seek(0)
+from remote_pdb import RemotePdb
 
 try:
     from io import BytesIO
 except ImportError:
     from StringIO import StringIO as BytesIO
 
+# breaker = RemotePdb('0.0.0.0', 4444)
+
 def base_url():
     return "http://localhost/irods-rest/0.9.0/"
 
-if [int(x) for x in pycurl.version_info()[1].split('.')][:2] <= [7,29]:
-    StringIO = old_pycurl_mock_StringIO
-    BytesIO = old_pycurl_mock_BytesIO
-
 def authenticate(_user_name, _password, _auth_type):
-    buffer = StringIO()
+    buffer = BytesIO()
 
     creds = _user_name + ':' + _password
     buff  = creds.encode('ascii')
     token = base64.b64encode(buff , None)
 
     c = pycurl.Curl()
-    c.setopt(pycurl.HTTPHEADER,['Authorization: Native '+token])
+    c.setopt(pycurl.HTTPHEADER, ['Authorization: Native '+ token.decode()])
```

Comment:
other places we're leaving the spaces out... not sure if we have a strong convention...